client/src/components/Map/Map.tsx

```diff
@@ -0,0 +1,174 @@
+  const [state, dispatch] = useReducer(reducer, initialState);
+  const locationRef = useRef(null);
+  const wrapperRef = useRef(null);
+  const autosuggestUrl = 'https://geodata.nationaalgeoregister.nl/locatieserver/v3/suggest?fq=gemeentenaam:amsterdam&fq=type:adres&fl=id,weergavenaam,type,score,lat,lon&q=';
+  const locationUrl = 'https://geodata.nationaalgeoregister.nl/locatieserver/revgeo?type=adres&rows=1&fl=id,weergavenaam,straatnaam,huis_nlt,postcode,woonplaatsnaam,centroide_ll&distance=50&';
+
+  const { mapInstance, url, query, autosuggest, latLng, location, showAutosuggest} = state;
+
+  const fetchAutosuggest = async (url) => {
+    const response = await axios.get(url);
+    dispatch({
+      type: 'setAutosuggest',
+      payload: {
+        autosuggest: response.data?.response?.docs
+      }
+    })
+  };
+
+  const fetchLocation = async (location: any) => {
+    const response = await axios.get(`${locationUrl}&lat=${location.lat}&lon=${location.lng}`);
+    dispatch({
+      type: 'setLocation',
+      payload: {
+        location: response.data?.response?.docs
+      }
+    });
+  };
+
+  const onMapclick = (e: any) => {
+    dispatch({
+      type: 'setLatLng',
+      payload: {
+        latLng: e.latlng
+      }
+    });
+
+    fetchLocation(e.latlng);
+  }
+
+  const onAutosuggestClick = async (e: any, location: any) => {
+    e.preventDefault();
+    if (location.weergavenaam) {
+      locationRef.current.value = location.weergavenaam;
+      dispatch({
+        type: 'setShowAutosuggest',
+        payload: {
+          showAutosuggest: false
+        }
+      });
+    }
+
+    const response = await axios.get(`https://geodata.nationaalgeoregister.nl/locatieserver/v3/lookup?id=${location.id}`)
+    if (mapInstance && response.data.response.docs[0]) {
+      const loc = response.data.response.docs[0].centroide_ll.replace(/POINT\(|\)/, '').split(' ');
+      const flyTo = { lat: parseFloat(loc[1]), lng: parseFloat(loc[0]) };
+      mapInstance.flyTo(flyTo, 11);
+      dispatch({
+        type: 'setLatLng',
+        payload: {
+          latLng: flyTo
+        }
+      });
+    }
+  }
+
+  useEffect(() => {
+    fetchAutosuggest(url);
+  }, [url]);
+
+  useEffect(() => {
+    if (location && location.length) {
+      locationRef.current.value = location[0].weergavenaam;
+    } else if (location && location.length === 0) {
+      locationRef.current.value = '';
+    }
+  }, [location, locationRef]);
+
+  const handleClickOutside = event => {
+    if (wrapperRef.current && !wrapperRef.current.contains(event.target)) {
+      dispatch({
+        type: 'setShowAutosuggest',
+        payload: {
+          showAutosuggest: false
+        }
+      });
+
+      locationRef.current.value = '';
+    }
+  };
+  useEffect(() => {
+    document.body.addEventListener('click', handleClickOutside, false)
+  }, []);
+
+  return (
+    <div style={{ position: 'relative', width: '100%', height: '500px' }}>
```

Comment:
IPV deze inline style zou ik een styled component definieren. Is iets consistenter